pkg/bosh/manifest/instance_group.go

```diff
@@ -25,22 +27,72 @@ func (instanceGroups InstanceGroups) InstanceGroupByName(name string) (*Instance
 
 // InstanceGroup from BOSH deployment manifest.
 type InstanceGroup struct {
-	Name               string                 `json:"name"`
-	Instances          int                    `json:"instances"`
-	AZs                []string               `json:"azs"`
-	Jobs               []Job                  `json:"jobs"`
-	VMType             string                 `json:"vm_type,omitempty"`
-	VMExtensions       []string               `json:"vm_extensions,omitempty"`
-	VMResources        *VMResource            `json:"vm_resources"`
-	Stemcell           string                 `json:"stemcell"`
-	PersistentDisk     *int                   `json:"persistent_disk,omitempty"`
-	PersistentDiskType string                 `json:"persistent_disk_type,omitempty"`
-	Networks           []*Network             `json:"networks,omitempty"`
-	Update             *Update                `json:"update,omitempty"`
-	MigratedFrom       []*MigratedFrom        `json:"migrated_from,omitempty"`
-	LifeCycle          InstanceGroupType      `json:"lifecycle,omitempty"`
-	Properties         map[string]interface{} `json:"properties,omitempty"`
-	Env                AgentEnv               `json:"env,omitempty"`
+	Name               string                  `json:"name"`
+	Instances          int                     `json:"instances"`
+	AZs                []string                `json:"azs"`
+	Jobs               []Job                   `json:"jobs"`
+	VMType             string                  `json:"vm_type,omitempty"`
+	VMExtensions       []string                `json:"vm_extensions,omitempty"`
+	VMResources        *VMResource             `json:"vm_resources"`
+	Stemcell           string                  `json:"stemcell"`
+	PersistentDisk     *int                    `json:"persistent_disk,omitempty"`
+	PersistentDiskType string                  `json:"persistent_disk_type,omitempty"`
+	Networks           []*Network              `json:"networks,omitempty"`
+	Update             *Update                 `json:"update,omitempty"`
+	MigratedFrom       []*MigratedFrom         `json:"migrated_from,omitempty"`
+	LifeCycle          InstanceGroupType       `json:"lifecycle,omitempty"`
+	Properties         InstanceGroupProperties `json:"properties,omitempty"`
+	Env                AgentEnv                `json:"env,omitempty"`
+}
+
+// InstanceGroupQuarks represents the quark property of a InstanceGroup
+type InstanceGroupQuarks struct {
+	RequiredService *string `json:"required_service,omitempty"`
+}
+
+// InstanceGroupProperties represents the properties map of a InstanceGroup
+type InstanceGroupProperties struct {
```

Comment:
Can this be solved like this https://github.com/cloudfoundry-incubator/cf-operator/blob/master/pkg/bosh/manifest/job.go#L63-L66

I kind of remember that we don't need the JSON workaround with `sigs.k8s.io/yaml`.